internal/resources/configuration_profile_teams.go

```diff
@@ -0,0 +1,346 @@
+	api *api.API
+}
+
+func (r *configurationProfileTeamsResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_configuration_profile_teams"
+}
+
+func (r *configurationProfileTeamsResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: `Manage the Microsoft Teams configuration profile.
+
+The profile is global, adding multiple resources of this kind will cause them to override each other.
+`,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The GraphQL Node ID of the configuration profile.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"profile": schema.StringAttribute{
+				Description: "The profile name.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"webhook": schema.ListNestedBlock{
+				Description: "Microsoft Teams webhook configuration.",
+				Validators: []validator.List{
+					listvalidator.SizeAtLeast(1),
+					listvalidator.IsRequired(),
+				},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"name": schema.StringAttribute{
+							Description: "The webhook name.",
+							Required:    true,
+						},
+						"url": schema.StringAttribute{
+							Description: "The encrypted webhook URL returned by the API.",
+							Computed:    true,
+						},
+						"url_wo": schema.StringAttribute{
+							Description: "The input value for the webhook URL.",
+							Required:    true,
+							Sensitive:   true,
+							WriteOnly:   true,
+						},
+						"url_wo_version": schema.StringAttribute{
+							Description: "The version for the webhook URL. Must be changed to update url_wo.",
+							Required:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *configurationProfileTeamsResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if pd, err := providerdata.GetResourceProviderData(req); err != nil {
+		errors.AddDiagError(&resp.Diagnostics, err)
+	} else if pd != nil {
+		r.api = pd.API
+	}
+}
+
+func (r *configurationProfileTeamsResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state models.ConfigurationProfileTeamsResource
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	whSecretsState, diags := r.getWebhooksSecrets(ctx, state)
+	resp.Diagnostics.Append(diags...)
+
+	teamsConfig, err := r.api.ConfigurationProfile.ReadTeams(ctx)
+	if err != nil {
+		handleAPIError(ctx, &resp.State, &resp.Diagnostics, err)
+		return
+	}
+
+	names := models.ListItemsIdentifiers(state.Webhooks, "name")
```

Comment:
maybe an explicit comment about how we're restoring the input ordering here?

(…and _maybe_ this whole repeated block deserves a name of its own for that comment to hang off?)